### PR TITLE
Add a Conflict message option that has requirement trees

### PR DESCRIPTION
**Pull request by segiddins**
Just like in Bundler!

### CODE REVIEW

**Comment by segiddins** [comment]
@DanToml please review?

File: spec/resolver_spec.rb [hunk 1]
```diff
@@ -156,9 +160,45 @@ def self.save!(path, name, index, requirements, resolved)
       end
 
       it 'includes the source of a user-specified unsatisfied dependency' do
-        expect do
-          @resolver.resolve([Gem::Dependency.new('missing', '3.0')], DependencyGraph.new)
-        end.to raise_error(VersionConflict, /required by `user-specified dependency`/)
+        expect { @resolver.resolve([Gem::Dependency.new('missing', '3.0')], DependencyGraph.new) }.
+          to raise_error(VersionConflict) do |conflict|
+          expect(conflict.message).to eq <<-EOS.strip
```
Comment (greysteil) [inline spec/resolver_spec.rb]: Indent this line, down to line 175, by another two spaces.

Comment (segiddins) [inline spec/resolver_spec.rb]: rubocop says no

Comment (greysteil) [inline spec/resolver_spec.rb]: Weird - that looks like a bug in rubocop to me. Meh. 👍  then.

**Pull request merged**